### DISCUSSION TITLE
Notmuch layer's keybindings

### DISCUSSION THREAD

**Discussion by agenbite** [syl20bnr/spacemacs#14582] [Q&A]
Hi, everyone,

I'm using the notmuch layer and I tend to prefer seeing my inbox in tree mode. However, this prevents me from using the d/D bindings to delete messages. I get a "Nothing to tag!" response. I don't know if this is a bug or the expected behaviour. If it's a bug, I'd like to help in solving it. If it is not, it'd be great if someone could point me in the right direction to add the functionality I'm missing.

I hope I'm asking in the right place.

Regards,

**Comment by lebensterben** [comment]
strange

https://github.com/syl20bnr/spacemacs/blob/8497f5ef8ee1feb7e514a498e5f66044d9ebb665/layers/%2Bemail/notmuch/packages.el#L130

Tree mode map clearly defines d/D as deletions.

This is the where you should start debugging

https://github.com/syl20bnr/spacemacs/blob/8497f5ef8ee1feb7e514a498e5f66044d9ebb665/layers/%2Bemail/notmuch/funcs.el#L47

**Reply by agenbite** [reply]
Thanks, lebensterben! The fact is that a/A do work in tree mode, whilst they're not actually defined in packages.el, and this obviously means there's some other conflicting configuration somewhere. Where should I start looking for this conflict, I don't know. Any help is welcome.

**Reply by lebensterben** [reply]
I already pointed out.

**Reply by agenbite** [reply]
Thanks again, I did read your message, however I can't seem to find any way out. 

I understand that all the bindings for `notmuch-tree` mode are defined from line 125, whereas all the bindings for `notmuch-search` mode are defined from line 135. 

However, in the bindings for `notmuch-search` I see the definition of bindings d/D and a/A, and they both work fine. However, for `notmuch-tree`, only d/D are explicitly defined (in lines 130:131, as you pointed out). However, surprisingly, d/D do not work in `notmuch-tree`, whereas a/A do. 

I've installed the layer altogether anew, and the problem remains. In my dotfile there's no definition of bindings apart from 
```
'(notmuch-tagging-keys
  (quote
    (("a" notmuch-archive-tags "Archive")
     ("u" notmuch-show-mark-read-tags "Mark read")
     ("s" ("+spam" "-inbox") "Mark as spam")
     ("d" ("+deleted" "-inbox") "Delete"))))
```
which allow me to delete in notmuch-tree using two keystrokes instead of one.

I'm really at a loss here.

**Reply by agenbite** [reply]
Ok, I found the problem (and apparently solved it). I proposed a PR with this: #14583
